src/main/helm/helm-service.ts

```diff
@@ -95,21 +93,27 @@ class HelmService {
 
     return { message: output };
   }
+}
 
-  protected excludeDeprecated(entries: any) {
-    for (const key in entries) {
-      entries[key] = entries[key].filter((entry: any) => {
-        if (Array.isArray(entry)) {
-          return entry[0]["deprecated"] != true;
-        }
+class ChartGroups {
+  private items: Map<string, HelmChart[]>;
 
-        return entry["deprecated"] != true;
-      });
-    }
+  constructor(groups: HelmChartGroups) {
+    this.items = new Map(Object.entries(groups));
+    this.excludeDeprecatedGroups();
+  }
 
-    return entries;
+  excludeDeprecatedGroups() {
```

Comment:
This should be private and probably just part of the getter

Reply:
Set as `private` but not sure about putting it into getter since I find it unexpected - getters shouldn't do any modifitations to entities. Function should do only one thing at best.